aws-api-appsync/src/main/java/com/amplifyframework/datastore/appsync/ModelWithMetadataDeserializer.java

```diff
@@ -41,8 +44,64 @@
             throw new JsonParseException("Expected a parameterized type during ModelWithMetadata deserialization.");
         }
 
-        Model model = context.deserialize(json, modelClassType);
-        ModelMetadata metadata = context.deserialize(json, ModelMetadata.class);
-        return new ModelWithMetadata<>(model, metadata);
+        if (modelClassType == SerializedModel.class) {
+            JsonObject jsonObject = json.getAsJsonObject();
+            String id = null;
+            int version = 0;
+            Long changedAt = null;
+            Boolean deleted = null;
+            String modelName = null;
+            Map<String, Object> serializedData = new HashMap<>();
+            for (Map.Entry<String, JsonElement> entry : jsonObject.entrySet()) {
+                switch (entry.getKey()) {
+                    case "id":
+                        id = entry.getValue().isJsonNull() ? null : entry.getValue().getAsString();
+                        break;
+                    case "_version":
+                        version = entry.getValue().isJsonNull() ? null : entry.getValue().getAsInt();
+                        break;
+                    case "_lastChangedAt":
+                        changedAt = entry.getValue().isJsonNull() ? null : entry.getValue().getAsLong();
+                        break;
+                    case "_deleted":
+                        deleted = entry.getValue().isJsonNull() ? null : entry.getValue().getAsBoolean();
+                        break;
+                    case "modelName":
+                        modelName = entry.getValue().isJsonNull() ? null : entry.getValue().getAsString();
+                        break;
+                    default :
+                        serializedData.put(entry.getKey(), getJavaPrimitiveFromJson(entry.getValue()));
+                }
+            }
+            ModelMetadata metadata = new ModelMetadata(id, deleted, version, changedAt);
```

Comment:
It would be best to create the `ModelMetadata` in the same way for all scenarios, like this:

```
ModelMetadata metadata = context.deserialize(json, ModelMetadata.class);
```

The benefit of using the `JsonDeserializationContext` where possible is that it has the context for how to handle all of the various types defined in `GsonFactory.withDeserializers`.  For example, it knows how to deserialize dates that are represented as Strings in Java to the concrete types (e.g. `Temporal.Timestamp`).  Right now, `ModelMetadata.changedAt` is just a `Long`, but I'm actually planning to change it to a `Temporal.Timestamp`, since that is how it is represented in AppSync.  Once I make that change, we will want need to make sure it is deserialized by the `JsonDeserializationContext`.

Reply:
Makes sense, I thought about using the `JsonDeserializationContext` but didn't figure out how to get the _rest of the fields_ but I guess I can just skip over the metadata fields if I use `JsonDeserializationContext`